controllers/authpolicy_controller.go

```diff
@@ -290,6 +305,59 @@ func (r *AuthPolicyReconciler) removeAuthSchemes(ctx context.Context, ap *kuadra
 	return nil
 }
 
+func (r *AuthPolicyReconciler) desiredAuthConfig(ctx context.Context, ap *kuadrantv1beta1.AuthPolicy) (*authorinov1beta1.AuthConfig, error) {
+	hosts, err := r.policyHosts(ctx, ap)
+	if err != nil {
+		return nil, err
+	}
+
+	return &authorinov1beta1.AuthConfig{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "AuthConfig",
+			APIVersion: authorinov1beta1.GroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      authConfigName(client.ObjectKeyFromObject(ap)),
+			Namespace: common.KuadrantNamespace,
+		},
+		Spec: authorinov1beta1.AuthConfigSpec{
+			Hosts:         hosts,
+			Patterns:      ap.Spec.AuthScheme.Patterns,
+			Conditions:    ap.Spec.AuthScheme.Conditions,
+			Identity:      ap.Spec.AuthScheme.Identity,
+			Metadata:      ap.Spec.AuthScheme.Metadata,
+			Authorization: ap.Spec.AuthScheme.Authorization,
+			Response:      ap.Spec.AuthScheme.Response,
+			DenyWith:      ap.Spec.AuthScheme.DenyWith,
+		},
+	}, nil
+}
+
+func (r *AuthPolicyReconciler) policyHosts(ctx context.Context, ap *kuadrantv1beta1.AuthPolicy) ([]string, error) {
+	if len(ap.Spec.AuthRules) == 0 {
+		return r.TargetHostnames(ctx, ap.Spec.TargetRef, ap.Namespace)
+	}
+
+	uniqueHostnamesMap := make(map[string]interface{})
+	for idx := range ap.Spec.AuthRules {
+		if len(ap.Spec.AuthRules[idx].Hosts) == 0 {
+			// When one of the rules does not have hosts, just return target hostnames
+			return r.TargetHostnames(ctx, ap.Spec.TargetRef, ap.Namespace)
```

Comment:
So what happens when the next `idx` of the `AuthRules` has _Hosts_ ? those would be ignored right? and used the _TargetHostnames_